--- CategoryPage.php.orig	2009-07-20 23:03:36.000000000 +0200
+++ CategoryPage.php	2009-07-20 23:48:40.000000000 +0200
@@ -189,11 +189,11 @@
 	 */
 	function addPage( $title, $sortkey, $pageLength, $isRedirect = false ) {
 		global $wgContLang;
-		$titletext = $wgContLang->convert( $title->getPrefixedText() );
+		$titletext = $wgContLang->convert( $title->getText() );
 		$this->articles[] = $isRedirect
 			? '<span class="redirect-in-category">' . $this->getSkin()->makeKnownLinkObj( $title, $titletext ) . '</span>'
 			: $this->getSkin()->makeSizeLinkObj( $pageLength, $title, $titletext );
-		$this->articles_start_char[] = $wgContLang->convert( $wgContLang->firstChar( $sortkey ) );
+		$this->articles_start_char[] = $wgContLang->convert( $wgContLang->firstChar( $titletext ) );
 	}
 
 	function finaliseCategoryState() {
@@ -223,7 +223,7 @@
 				'cat_id', 'cat_title', 'cat_subcats', 'cat_pages', 'cat_files' ),
 			array( $pageCondition, 'cl_to' => $this->title->getDBkey() ),
 			__METHOD__,
-			array( 'ORDER BY' => $this->flip ? 'cl_sortkey DESC' : 'cl_sortkey',
+			array( 'ORDER BY' => $this->flip ? 'page_title DESC' : 'page_title',
 				'USE INDEX' => array( 'categorylinks' => 'cl_sortkey' ),
 				'LIMIT'    => $this->limit + 1 ),
 			array( 'categorylinks'  => array( 'INNER JOIN', 'cl_from = page_id' ),
