--- CategoryPage.php.orig	2010-08-09 18:36:38.000000000 +0200
+++ CategoryPage.php	2010-08-09 18:42:33.000000000 +0200
@@ -207,12 +207,12 @@
 			? '<span class="redirect-in-category">' .
 				$this->getSkin()->link(
 					$title,
-					null,
+					$title->getText(),
 					array(),
 					array(),
 					array( 'known', 'noclasses' )
 				) . '</span>'
-			: $this->getSkin()->makeSizeLinkObj( $pageLength, $title );
+			: $this->getSkin()->makeSizeLinkObj( $pageLength, $title, $title->getText() );
 		$this->articles_start_char[] = $wgContLang->convert( $wgContLang->firstChar( $sortkey ) );
 	}
 
