--- CategoryPage.php.orig	2008-03-21 20:15:38.000000000 +0100
+++ CategoryPage.php	2008-03-21 20:15:50.000000000 +0100
@@ -186,7 +186,7 @@
 		$this->articles[] = $isRedirect
 			? '<span class="redirect-in-category">' . $this->getSkin()->makeKnownLinkObj( $title ) . '</span>'
 			: $this->getSkin()->makeSizeLinkObj( $pageLength, $title );
-		$this->articles_start_char[] = $wgContLang->convert( $wgContLang->firstChar( $sortkey ) );
+		$this->articles_start_char[] = $wgContLang->convert( $wgContLang->firstChar( $title->getText() ) );
 	}
 
 	function finaliseCategoryState() {
@@ -219,7 +219,7 @@
 			       #'page_is_redirect' => 0),
 			#+ $pageCondition,
 			__METHOD__,
-			array( 'ORDER BY' => $this->flip ? 'cl_sortkey DESC' : 'cl_sortkey',
+			array( 'ORDER BY' => $this->flip ? 'page_title DESC' : 'page_title',
 			       'USE INDEX' => 'cl_sortkey', 
 			       'LIMIT'    => $this->limit + 1 ) );
 
